libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/prompts/UserTokenAccess.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MT License.
+
+package com.microsoft.bot.dialogs.prompts;
+
+import com.microsoft.bot.builder.ConnectorClientBuilder;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.connector.ConnectorClient;
+import com.microsoft.bot.connector.authentication.ClaimsIdentity;
+import com.microsoft.bot.connector.authentication.ConnectorFactory;
+import com.microsoft.bot.connector.authentication.UserTokenClient;
+import com.microsoft.bot.schema.SignInResource;
+import com.microsoft.bot.schema.TokenExchangeRequest;
+import com.microsoft.bot.schema.TokenResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class UserTokenAccess {
+
+    private UserTokenAccess() {
+    }
+
+    public static CompletableFuture<TokenResponse> getUserToken(
+        TurnContext turnContext,
+        OAuthPromptSettings settings,
+        String magicCode) {
+
+        UserTokenClient userTokenClient = turnContext.getTurnState().get(UserTokenClient.class);
+
+        if (userTokenClient != null) {
+            return userTokenClient.getUserToken(
+                turnContext.getActivity().getFrom().getId(),
+                settings.getConnectionName(),
+                turnContext.getActivity().getChannelId(),
+                magicCode);
+        } else {
+            throw new UnsupportedOperationException("OAuth prompt is not supported by the current adapter");
```

Comment:
I was wondering if the error describes the actual behavior. What I understand is just that an instance of `UserTokenClient` cannot be found. The exception text in C# is there because there is an extra condition that was not migrated.

Reply:
good point. I also found the message not descriptive enough